standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/AbstractJDBCConnectorProvider.java

```diff
@@ -129,8 +129,21 @@ protected Connection getConnection() {
     throw new RuntimeException("unexpected type for connection handle");
   }
 
+  protected boolean isOpen() {
+    if (handle == null) {
+      return false;
+    }
+    try {
+      if (handle instanceof Connection)
+        return ((Connection) handle).isValid(3);
+    } catch (SQLException e) {
+      LOG.warn("Could not validate jdbc connection to "+ jdbcUrl, e);
+    }
+    return false;
+  }
+
   @Override public void close() {
-    if (isOpen) {
+    if (isOpen()) {
```

Comment:
though the connection may be invalid, but it may still remain opening, should we use `isClosed()` instead?

Reply:
Make sense. Renamed it to `isClosed()`.

Reply:
Sorry for misunderstanding what you meant. I wrap a new method `isClosed()` to check the connection` closed` status.